osu.Game/Beatmaps/BeatmapDifficultyCache.cs

```diff
@@ -249,49 +244,17 @@ private StarDifficulty computeDifficulty(in DifficultyCacheLookup key, BeatmapIn
                 if (rulesetInfo.Equals(beatmapInfo.Ruleset))
                 {
                     Logger.Error(e, $"Failed to convert {beatmapInfo.OnlineBeatmapID} to the beatmap's default ruleset ({beatmapInfo.Ruleset}).");
-                    return Cache[key] = new StarDifficulty();
+                    return new StarDifficulty();
                 }
 
-                // Check the cache first because this is now a different ruleset than the one previously guarded against.
-                if (tryGetExisting(beatmapInfo, beatmapInfo.Ruleset, Array.Empty<Mod>(), out var existingDefault, out var existingDefaultKey))
-                    return existingDefault;
-
-                return computeDifficulty(existingDefaultKey, beatmapInfo, beatmapInfo.Ruleset);
+                return GetAsync(new DifficultyCacheLookup(key.Beatmap, key.Beatmap.Ruleset, key.OrderedMods)).Result;
```

Comment:
Is this safe? We're already on the `updateScheduler` at this point, which has only one thread available to it. We can't be firing another async task.

Reply:
most scheduler implementations recognise this and don't spawn a new thread if already on the correct thread/pool.

![image](https://user-images.githubusercontent.com/191335/98340132-3c88c180-2050-11eb-9554-814a690d7c10.png)

It's implemented correctly in `ThreadedTaskScheduler` so shouldn't be an issue. Worth confirming this though.